### PR TITLE
SC-2299 Refactoring Location Module

### PR DESCRIPTION
**Pull request by reshmi-nair**

### CODE REVIEW

File: service/src/main/java/org/sunbird/actor/location/LocationActor.java [hunk 1]
```diff
@@ -86,7 +87,7 @@ private void createLocation(Request request) {
             String id = ProjectUtil.generateUniqueId();
             locationRequest.setId(id);
             Location location = mapper.convertValue(locationRequest, Location.class);
-            Response response = locationDao.create(location, request.getRequestContext());
+            Response response = locationService.createLocation(location, request.getRequestContext());
```
Comment (amit-tarento) [inline service/src/main/java/org/sunbird/actor/location/LocationActor.java:90]: We should move all these Location creation logic to service layer

Comment (amit-tarento) [reply service/src/main/java/org/sunbird/actor/location/LocationActor.java:90]: Move validateUpsertLocnReq either in request validator class or in service layer

Comment (amit-tarento) [reply service/src/main/java/org/sunbird/actor/location/LocationActor.java:90]: Also move saveDataToES method to dao and call it via service layer

Comment (reshmi-nair) [reply service/src/main/java/org/sunbird/actor/location/LocationActor.java:90]: Resolving this. Not required to move this as other service calls will result in request validation each time

**Review by amit-tarento — APPROVED** [review]

**Pull request merged**